kentikapi/internal/synthetics/test_results.go

```diff
@@ -0,0 +1,287 @@
+package synthetics
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"time"
+
+	syntheticspb "github.com/kentik/api-schema-public/gen/go/kentik/synthetics/v202202"
+	"github.com/kentik/community_sdk_golang/kentikapi/internal/convert"
+	"github.com/kentik/community_sdk_golang/kentikapi/synthetics"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+func getResultsForTestsRequestToPayload(
+	req synthetics.GetResultsForTestsRequest,
+) *syntheticspb.GetResultsForTestsRequest {
+	return &syntheticspb.GetResultsForTestsRequest{
+		Ids:       req.TestIDs,
+		StartTime: timestamppb.New(req.StartTime),
+		EndTime:   timestamppb.New(req.EndTime),
+		AgentIds:  req.AgentIDs,
+		Targets:   convert.IPsToStrings(req.Targets),
+	}
+}
+
+type getResultsForTestsResponse syntheticspb.GetResultsForTestsResponse
+
+func (r *getResultsForTestsResponse) ToModel() ([]synthetics.TestResults, error) {
+	if r == nil {
```

Comment:
Changed: 6f3630033fddcd2de2fa3e849733bc67044b10ac

Note that this is a corner case when response object is not initialized. In current version of generated gRPC bindings it never happens, because response is initialized upfront: https://github.com/kentik/api-schema-public/blob/896729e59945404959e930f63f7a8965c109537d/gen/go/kentik/synthetics/v202202/synthetics_grpc.pb.go#L35

But I changed it anyway to be on the safe side.
Note that I return nil slices, because there is no need to initialize them - they work correctly with len(), append() and so on.